src/App.tsx

```diff
@@ -1,24 +1,233 @@
+      ...todo,
+      completed: !todo.completed,
+    }));
+  };
+
+  const handleAddTodo = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!todoTitle.trim()) {
+      showError(ErrorMessage.emptyTitle);
+
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: Math.max(...todos.map(todo => todo.id)) + 1,
+      title: todoTitle,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setTempTodo({ ...newTodo, id: 0 });
+
+    setProcessingIds(current => [...current, 0]);
+    createTodo(newTodo)
+      .then(newToDo => {
+        setTodos(currentTodos => [...currentTodos, newToDo]);
+        resetField();
+      })
+      .catch(() => {
+        setTempTodo(null);
+        showError(ErrorMessage.addError);
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setProcessingIds([]);
+      });
+
+    resetField();
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(data => {
+        const todosData = data as Todo[];
+
+        setTodos(todosData);
+      })
+      .catch(() => {
+        showError(ErrorMessage.loadError);
+      });
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            type="button"
+            aria-label="button"
+            onClick={handleToggleAll}
+            className={
+              cn(
+                'todoapp__toggle-all',
+                {
+                  active: isAllCompleted,
+                },
+              )
+            }
+          />
+
+          <form onSubmit={handleAddTodo}>
+            <input
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={todoTitle}
+              onChange={event => setTodoTitle(event.target.value)}
+              disabled={!!tempTodo}
+            />
+          </form>
+        </header>
+
+        {tempTodo
+          ? (
+            <TodoList
+              todos={[...filteredTodos, tempTodo]}
+              onDelete={handleDeleteTodo}
+              onUpdate={handleUpdateTodo}
+              processingIds={processingIds}
+            />
+          )
+          : (
+            <TodoList
+              todos={filteredTodos}
+              onDelete={handleDeleteTodo}
+              onUpdate={handleUpdateTodo}
+              processingIds={processingIds}
+            />
+          )}
```

Comment:
better to pass tempTodo as props and make proper logic inside component instead doing it by ternary operator with component